patches/tModLoader/Terraria/ModLoader/ModPlayer.cs

```diff
@@ -914,5 +895,49 @@ public abstract class ModPlayer : ModType
 		/// <param name="mediumCoreDeath">Whether you are setting up a mediumcore player's inventory after their death.</param>
 		public virtual void ModifyStartingInventory(IReadOnlyDictionary<string, List<Item>> itemsByMod, bool mediumCoreDeath) {
 		}
+
+		public struct Regen
+		{
+			/// <summary>
+			/// Corresponds to incrementing or decrementing player.xRegen. The player will recover life/mana at a rate of half the number you add per second. 
+			/// </summary>
+			public float deltaRegen;
+			/// <summary>
+			/// Corresponds to incrementing or decrementing player.LifeRegenTime or player.manaRegenDelayBonus depending on context. Changes the speed at which the player reaches its maximum natural life regeneration or time to begin regenerating mana.
+			/// </summary>
+			public float deltaRate;
+			/// <summary>
+			/// Allows you to modify the power of the player's natural life/mana regeneration. This can be done by setting a multiplication value. For example, campfires multiply it by 1.1, while walking multiplies it by 0.5.
+			/// </summary>
+			public float multiplyRegen;
+			/// <summary>
+			/// Allows you to force no positive regeneration to occur. For example, the "On Fire!" debuff prevents positive life regenaration.
+			/// </summary>
+			public bool maxZeroRegen;
+
+			public static Regen Create() {
+				return new Regen { deltaRegen = 0, deltaRate = 0, multiplyRegen = 1, maxZeroRegen = false };
+			}
+			public void ResetVals() {
+				deltaRate = 0;
+				deltaRegen = 0;
+				multiplyRegen = 1;
+				maxZeroRegen = false;
+			}
+		}
```

Comment:
class should have a `Combine` method like `DamageModifier`

Reply:
Per original post, couldn't find that class. Was it changed or removed in the damage class rework?
Regardless, combine method is self-explanatory - combine the structs. 
Only thing I miss is name consistency : (